### DISCUSSION TITLE
allow displacy to display several kind of spans

### DISCUSSION THREAD

**Discussion by joseberlines** [explosion/spaCy#11311] [Help: Other Questions] [answered]
I would like to display several categories of spans
going to the documentation I can see:
https://spacy.io/usage/visualizers#span

```
import spacy
from spacy import displacy
from spacy.tokens import Span

text = "Welcome to the Bank of China. We are at home."

nlp = spacy.blank("en")
doc = nlp(text)

doc.spans["sc"] = [
    Span(doc, 3, 6, "ORG"), 
    Span(doc, 5, 6, "GPE"),
]

# adding other custom spans
doc.spans["custom"] = [Span(doc, 9, 10, "home")]
options = {"spans_key": "custom"}
displacy.serve(doc, style="span", options=options)
```

with
`options = {"spans_key": ["custom","sc"]}`
it does not work.

It looks like it is only possible to visualize ONE category of spans.

is that so? if not this is a suggestion for a change. (I would not call it improvement because there might be a reason for this way of working)

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @joseberlines , 

Yes, the visualizer can only display one `spans_key` at a time. It might be tricky to distinguish between two keys in the visualization, hence the design. If you like to show both, you can combine them into a single `spans_key` (e.g., `custom_sc`) and pass that value to displaCy.

**Reply by joseberlines** [reply]
Cool. Indeed when visualizing I assume there won't be needed to visualize a million spans. So combining all might be an idea.

**Comment by gonconist** [comment]
Hi,

What about the case where we want to visualize both gold and predicted spans in the same plot using different colors? The gold span can match the predicted one, but nevertheless, I'd like to use a different color to make the distinction.

Any suggestions?
